d3a_api_client/grid_fee_calculation.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from copy import copy
+
+from d3a_interface.utils import key_in_dict_and_not_none
+
+
+class GridFeeCalculation:
+
+    def __init__(self):
+        self.latest_grid_stats_tree = {}
+        self.paths_to_root_mapping = {}
+        self.market_name_grid_fee_mapping = {"last_market_fee": {},
+                                             "next_market_fee": {}}
+
+    def _handle_grid_stats(self, message):
+        for device_event in message["content"]:
+            if device_event["event"] == "grid stats":
+                self.latest_grid_stats_tree = device_event["grid_stats_tree"]
+                message["content"].remove(device_event)
+        self._populate_grid_fee_mappings()
+
+    def _populate_grid_fee_mappings(self):
+        for fee_type in ["last_market_fee", "next_market_fee"]:
+            self._get_grid_fee_area_mapping_from_dict(self.latest_grid_stats_tree, fee_type)
+
+        self._get_all_paths_in_grid_stats_dict(self.latest_grid_stats_tree, [])
+
+    def _get_grid_fee_area_mapping_from_dict(self, indict, fee_type):
+        for child_name, child_stats in indict.items():
+            if "children" in child_stats and fee_type in child_stats:
+                self.market_name_grid_fee_mapping[fee_type][child_name] = child_stats[fee_type]
+                self._get_grid_fee_area_mapping_from_dict(child_stats["children"], fee_type)
+
+    def _get_all_paths_in_grid_stats_dict(self, indict, parent_path):
+        for child_name, child_stats in indict.items():
+            sub_path = parent_path + [child_name]
+            self.paths_to_root_mapping[child_name] = parent_path
+            if "children" in child_stats:
+                self._get_all_paths_in_grid_stats_dict(child_stats["children"], sub_path)
+
+    @staticmethod
+    def _strip_away_intersection_from_list(in_list, intersection):
+        out_list = copy(in_list)
+        for inter in intersection:
+            out_list.remove(inter)
+        return out_list
```

Comment:
I am not sure which of the 2 is better WRT performance, but if you want to convert it to an one-liner, there is a set method calculating this:
```suggestion
        return list(set(in_list).summetric_difference(set(intersection)))
```

Reply:
Will do, looks better IMO.